python/tests/core.py

```diff
@@ -0,0 +1,72 @@
+from vw_executor.vw import Vw
+from vw_executor.vw_opts import Grid
+from numpy.testing import assert_allclose
+import pandas as pd
+import numpy as np
+import pytest
+import os
+from test_helper import json_to_dict_list, dynamic_function_call, get_function_object, generate_string_combinations
+
+CURR_DICT = os.path.dirname(os.path.abspath(__file__))
+
+def combine_list_cmds_grids(cmds, base_grid):
```

Comment:
seems like a lot of it is already implemented as __add__/__mul__ operators of Grid and can be reused